src/js/control/textarea.tinymce.js

```diff
@@ -70,36 +71,45 @@ export default class controlTinymce extends controlTextarea {
 
   /**
    * When the element is rendered into the DOM, execute the following code to initialise it
-   * @param {Object} evt - event
    */
-  onRender(evt) {
-    if (window.tinymce.editors[this.id]) {
-      window.tinymce.editors[this.id].remove()
+  onRender() {
+    const oldInst = window.tinymce.get(this.id)
+    if (oldInst) {
+      window.tinymce.remove(oldInst)
     }
 
     // define options & allow them to be overwritten in the class config
     const options = jQuery.extend(this.editorOptions, this.classConfig)
     options.target = this.field
 
-    setTimeout(() => {
-      // initialise the editor
-      window.tinymce.init(options)
-    }, 100)
-
-    // Set userData
-    if (this.config.userData) {
-      window.tinymce.editors[this.id].setContent(this.parsedHtml(this.config.userData[0]))
+    //Remove any defined plugins from the list if they have been removed or moved to Core as part of major version updates
+    const removedPlugins = []
+    if (Number(window.tinymce.majorVersion) >= 5) {
+      removedPlugins.push('contextmenu')
     }
+    if (Number(window.tinymce.majorVersion) >= 6) {
+      removedPlugins.push('paste','print')
+    }
+    options.plugins = options.plugins.filter(plugin => {
+      return (removedPlugins.indexOf(plugin) === -1)
+    })
 
-    if (window.lastFormBuilderCopiedTinyMCE) {
-      const timeout = setTimeout(() => {
-        window.tinymce.editors[this.id].setContent(this.parsedHtml(window.lastFormBuilderCopiedTinyMCE))
-        window.lastFormBuilderCopiedTinyMCE = null
-        clearTimeout(timeout)
-      }, 300)
+    const userData = this.config.userData ? this.parsedHtml(this.config.userData[0]) : undefined
+    const copiedData = window.lastFormBuilderCopiedTinyMCE ? this.parsedHtml(window.lastFormBuilderCopiedTinyMCE) : undefined
+    window.lastFormBuilderCopiedTinyMCE = null
+    const afterInit = function (inst) {
+      // Set userData
+      if (copiedData) {
+        inst.setContent(copiedData)
+      } else if (userData) {
+        inst.setContent(userData)
+      }
     }
 
-    return evt
+    setTimeout(() => {
+      // initialise the editor
+      window.tinymce.init(options).then(afterInit)
+    }, 0)
```

Comment:
do we still need this timeout?

Reply:
This was just to have the tinymce initialisation run async, slight benefit if you have many on the screen allow the browser to still draw and continue main execution rather than waiting for tinymce